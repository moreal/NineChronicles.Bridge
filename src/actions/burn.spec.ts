import { describe, it, expect } from "vitest";
import { encodeBurnAssetAction } from "./burn";
import { Address, RawPrivateKey } from "@planetarium/account";
import { encode } from "@planetarium/bencodex";

describe(encodeBurnAssetAction.name, async () => {
    const sender = Address.deriveFrom(await RawPrivateKey.generate().getPublicKey())
    const ncgMinter = Address.fromHex("47d082a115c63e7b58b1532d20e631538eafadde", true);
    describe("snapshot", () => {
        it("burn_asset with memo", () => {
            expect(encode(encodeBurnAssetAction(
                sender,
                {
                    currency: {
                        ticker: "NCG",
                        decimalPlaces: 0,
                        minters: new Set([
                            ncgMinter.toBytes()
                        ]),
                        maximumSupply: null,
                        totalSupplyTrackable: false,
                    },
                    rawValue: 100n,
                },
                "MEMO"
            ))).toMatchSnapshot();
        });
    });
});