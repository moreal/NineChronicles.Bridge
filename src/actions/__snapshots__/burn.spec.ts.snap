// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`encodeBurnAssetAction > snapshot > burn_asset with memo 1`] = `
Uint8Array [
  100,
  117,
  55,
  58,
  116,
  121,
  112,
  101,
  95,
  105,
  100,
  117,
  49,
  48,
  58,
  98,
  117,
  114,
  110,
  95,
  97,
  115,
  115,
  101,
  116,
  117,
  54,
  58,
  118,
  97,
  108,
  117,
  101,
  115,
  108,
  50,
  48,
  58,
  54,
  225,
  247,
  254,
  229,
  42,
  4,
  219,
  246,
  74,
  137,
  176,
  60,
  14,
  186,
  171,
  167,
  118,
  29,
  58,
  108,
  100,
  117,
  49,
  51,
  58,
  100,
  101,
  99,
  105,
  109,
  97,
  108,
  80,
  108,
  97,
  99,
  101,
  115,
  49,
  58,
  0,
  117,
  55,
  58,
  109,
  105,
  110,
  116,
  101,
  114,
  115,
  108,
  50,
  48,
  58,
  71,
  208,
  130,
  161,
  21,
  198,
  62,
  123,
  88,
  177,
  83,
  45,
  32,
  230,
  49,
  83,
  142,
  175,
  173,
  222,
  101,
  117,
  54,
  58,
  116,
  105,
  99,
  107,
  101,
  114,
  117,
  51,
  58,
  78,
  67,
  71,
  101,
  105,
  49,
  48,
  48,
  101,
  101,
  117,
  52,
  58,
  77,
  69,
  77,
  79,
  101,
  101,
]
`;
